### PR TITLE
ENG-7826, add startAppFlow() functionality

### PR DESCRIPTION
**Pull request by kishisaka-nid**
Advanced work will be put in another PR.

Call to the new startAppFlow: 

`NeuroID.getInstance()?.startAppFlow("form_zzzzz123")
`

Request sent to server: 

```
{
  "siteId": "form_picks709",
  "userId": "f3267106-18f3-434a-9733-07989d77afd4",
  "clientId": "9ce42cb2-45d6-46d0-8fbd-6139dae24d3d",
  "identityId": "f3267106-18f3-434a-9733-07989d77afd4",
  "registeredUserId": "",
  "pageTag": "com.sample.neuroid.us.activities.MainActivity",
  "pageId": "mobile",
  "tabId": "mobile",
  "responseId": "4d1360f7-0592-44ce-b48a-ea5a7ae82f82",
  "url": "android://com.sample.neuroid.us.activities.MainActivity",
  "jsVersion": "5.0.0",
  "sdkVersion": "5.android-3.2.1",
  "environment": "LIVE",
  "jsonEvents": [
    {
      "accel": {
        "x": -0.39780676,
        "y": 0.020338992,
        "z": 9.692726
    }
	.
	.
	Tons of events (SET_LINKED_SITE_ID event is below) 
	.
	.
    {
      "accel": {
        "x": -0.11964113,
        "y": -0.16091733,
        "z": 9.714262
      },
      "gyro": {
        "x": 7.635591E-4,
        "y": -1.5271181E-4,
        "z": -0.0
      },
      "ts": 1715644266938,
      "type": "SET_LINKED_SITE",
      "v": "form_zzzzz123"
    },
	.
	.
	more events, linkedSiteId at bottom! 
	.
	.
    {
      "accel": {
        "x": -0.39780676,
        "y": 0.020338992,
        "z": 9.677772
      },
      "attrs": [
        {
          "interval": "200sec"
        }
      ],
      "gyro": {
        "x": -7.635591E-4,
        "y": 0.0012216945,
        "z": -1.5271181E-4
      },
      "ts": 1715622026086,
      "type": "CADENCE_READING_ACCEL"
    }
  ],
  "linkedSiteId": "form_zzzzz123"
}
```

### CODE REVIEW

**Review by kevinsites — DISMISSED** [review]

**Review by cometman — DISMISSED** [review]

File: NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt [hunk 1]
```diff
@@ -1222,4 +1224,90 @@ open class NeuroIDClassUnitTests {
...
+            assertEquals("test_1234", flowResult.sessionID)
+            assertEquals(false, NeuroID.getInstance()?.isStopped())
+            verify {NeuroID.getInternalInstance()?.nidJobServiceManager?.sendEvents(
+                true)}
+            verify {NeuroID.getInternalInstance()?.dataStore?.saveEvent(NIDEventModel(
+                type= SET_LINKED_SITE, v="form_zzzzz123"))}
+            assertEquals("form_zzzzz123", NeuroID.linkedSiteID)
+        }
+        unmockkStatic(Calendar::class)
+    }
```
Comment (kevinsites) [inline NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt:1270]: this test might be named wrong. it says session not started but we assert that it is? I don't see us setting isSDKStarted to true prior to the test so presumably startSession is happening?

File: NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt [hunk 2]
```diff
@@ -1222,4 +1224,90 @@ open class NeuroIDClassUnitTests {
...
+            assertEquals(true, flowResult.started)
+            assertEquals("", flowResult.sessionID)
+            assertEquals(false, NeuroID.getInstance()?.isStopped())
+            verify {NeuroID.getInternalInstance()?.nidJobServiceManager?.sendEvents(
+                true)}
+            verify {NeuroID.getInternalInstance()?.dataStore?.saveEvent(NIDEventModel(
+                type= SET_LINKED_SITE, v="form_zzzzz123"))}
+            assertEquals("form_zzzzz123", NeuroID.linkedSiteID)
+        }
+        unmockkStatic(Calendar::class)
```
Comment (kevinsites) [inline NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt:1290]: same as the above, test names session_not_started but we check that it is

File: NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt [hunk 3]
```diff
@@ -339,6 +359,47 @@ class NeuroID
...
+                saveIntegrationHealthEvents()
+            }
+            // If not started then start
+            var startStatus: SessionStartResult = if (!isSDKStarted) {
+                // if userID passed then startSession else start
+                if (userID != null) {
+                    startSession(userID)
+                } else {
+                    start()
+                    SessionStartResult(isSDKStarted, "")
```
Comment (kevinsites) [inline NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt:384]: the result from start should be the first parameter in the SessionStartResult

File: NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt [hunk 4]
```diff
@@ -339,6 +359,47 @@ class NeuroID
...
+            // immediately flush events before anything else
+            CoroutineScope(Dispatchers.IO).launch {
+                nidJobServiceManager.sendEvents(true)
+                saveIntegrationHealthEvents()
+            }
+            // If not started then start
+            var startStatus: SessionStartResult = if (!isSDKStarted) {
+                // if userID passed then startSession else start
+                if (userID != null) {
+                    startSession(userID)
```
Comment (kevinsites) [inline NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt:381]: will this just return the result from startSession I am assuming? (not the most familiar with kotlin syntax)

Comment (kishisaka-nid) [reply NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt:381]: yep, just sets startStatus with the SessionStartResult from startSession call, start() or whatever else is returns a startSession and the failure case.

**Review by quacke-nid — APPROVED** [review]

**Review by kevinsites — APPROVED** [review]

**Pull request merged**